### ISSUE TITLE
[Bug]: Higgs Audio v3 decodes incomplete end frames as audio

### ISSUE THREAD

**Issue by ss1127201970** [vllm-project/vllm-omni#7941]
### Describe the bug

Higgs Audio v3 can produce a short noise burst at the end of a sentence. After reversing the delay pattern, the talker can leave two incomplete frames: one with EOC in codebook 0 and real codes in the other codebooks, followed by an EOC/pad frame.

Both stage-input adapters replace special/pad tokens with code 0 and trim only one final frame. The preceding mixed frame therefore reaches the codec as an unrelated, synthesized audio vector. In streaming mode, invalid frames can also enter the decoder as right context before the finish notification.

### Minimal reproducer

No model weights or GPU are needed; run in a vLLM-Omni environment:

```python
from types import SimpleNamespace
import torch
from vllm_omni.model_executor.stage_input_processors.higgs_audio_v3 import talker2code2wav

real = torch.arange(8 * 3).reshape(8, 3)
tail = torch.tensor([[1025, 1025]] + [[100 + q, 1025] for q in range(1, 7)] + [[107, -1]])
codes = torch.cat([real, tail], dim=1)
rows = torch.full((codes.shape[1] + 7, 8), 1024)
for q in range(8):
    rows[q:q + codes.shape[1], q] = codes[q]
output = SimpleNamespace(finished=True, outputs=[SimpleNamespace(multimodal_output={"codes": {"audio": rows}})])
result = talker2code2wav([output])[0]["prompt_token_ids"]
print(f"Expected 3 complete frames, got {len(result) // 8}")
assert result == real.reshape(-1).tolist()
```

On current main this returns 4 frames instead of 3. The expected result is exactly the three complete frames, with no EOC-to-zero replacement frame. A sequence containing only complete frames should also retain its final frame.

### Evidence and proposed fix

A captured synthesis contained 144 complete frames plus the two incomplete frames above. Current processing sent 145 frames to the codec. Decoding the same captured codes in FP32 with only the 144 complete frames reduced the final 20 ms RMS from 0.08618 to 0.000595 (about 43 dB).

Keep the contiguous valid prefix after de-delay in both adapters. For streaming, exclude incomplete frames from right context and derive emitted-frame/context counts from the retained prefix. This preserves frame coordinates and avoids dropping valid final audio.

### Environment / versions

- Ubuntu 22.04.5, Python 3.12, one RTX 4090 D (24 GB), driver 580.76.05
- vLLM 0.28.0, vLLM-Omni 0.28.0, PyTorch 2.13.0 / CUDA 13.0, Transformers 5.14.1
- The affected processor is also present at main `856d19f42cfccb1ae97aea1131b89662368baadb`.
- Model: `bosonai/higgs-audio-v3-tts-4b`, revision `239f63fb7b02b1aa085f98d9efae5e35cc5523e8`.
- Observed with voice cloning, seed 42, both full and chunked decoding; streaming settings: chunk 25, initial chunk 1, left context 25, right holdback 4.

AI disclosure: Codex assisted with diagnosis, the fix, regression tests, and this write-up.
